### PR TITLE
Add visual editor toggle to Whitehall

### PR DESCRIPTION
**Pull request by dnkrj**
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

# What
- Refactor govspeak editor component to make it more extensible
- Only show preview button when the user has javascript enabled
- Add a FlipFlop feature flag for the visual editor
- Use the feature flag to toggle the display of a visual editor button in the govspeak editor
- Hide the textarea when the button is clicked (does not display the visual editor yet)

# Why
- The first step of visual editor testing will be adding it as a button to the govspeak field
- https://trello.com/c/zG5zD2r2/2227-add-the-editor-to-whitehall

# Screenshot
Screenshot 2024-01-05 at 16 15 44

### CODE REVIEW

**Comment by dnkrj** [comment]
Updated the Jasmine tests and introduce a new test for the visual editor button

File: app/assets/javascripts/components/govspeak-editor.js [hunk 1]
```diff
@@ -47,63 +79,94 @@ window.GOVUK.Modules = window.GOVUK.Modules || {}
...
+      )
+    }
+  }
+
+  GovspeakEditor.prototype.showPreview = function (event) {
+    event.preventDefault()
+
+    this.backButton.classList.add('app-c-govspeak-editor__back-button--show')
+    this.previewButton.classList.remove(
+      'app-c-govspeak-editor__preview-button--show'
```
Comment (ryanb-gds) [inline app/assets/javascripts/components/govspeak-editor.js:97]: Would storing these class names in a variable, e.g. `showPreviewButtonClass`, make it any easier to read what's going on? I'm finding it tough to follow all the BEM class names, as you suggested I would

File: app/assets/stylesheets/components/_govspeak-editor.scss [hunk 2]
```diff
@@ -4,26 +4,32 @@
 
 .app-c-govspeak-editor__preview-button-wrapper {
   text-align: right;
+  display: flex;
+  justify-content: end;
+  gap: govuk-spacing(2);
```
Comment (ryanb-gds) [inline app/assets/stylesheets/components/_govspeak-editor.scss:9]: Lovely to see the `gap` attribute in the wild, was waiting in anticipation for that last time I did any serious frontend work

**Review by ryanb-gds — APPROVED** [review]
I see what you mean about the BEM classes! Maybe we ought to think about an abstraction there. This looks good for now though

File: app/assets/javascripts/components/govspeak-editor.js (hunk 1 above)
Comment (dnkrj) [reply app/assets/javascripts/components/govspeak-editor.js:97]: I think that I want to follow up with another PR on this, potentially introducing a slimmed down version of i-bem.js to whitehall or adding some context specific functions to this module for toggling visibility.

**Pull request merged**